src/app/pages/salidas/form-elementos-seleccionados/form-elementos-seleccionados.component.ts

```diff
@@ -40,149 +19,176 @@ import { isObject } from 'rxjs/internal-compatibility';
+    valueChanges.pipe(
+      startWith(''),
+      debounceTime(250),
+      distinctUntilChanged(),
+      switchMap((val) => this.loadDependencias(val) ),
+    ).subscribe((response: any) => {
+      this.dependencias = response.queryOptions[0].Id ? response.queryOptions : [];
+      this.getUbicaciones();
+    });
+  }
 
-    }
+  private filtroFuncionarios(nombre: string): TerceroCriterioContratista[] {
+    if (nombre.length >= 4 && Array.isArray(this.Funcionarios)) {
+      const valorFiltrado = nombre.toLowerCase();
+      return this.Funcionarios.filter(contr => this.muestraFuncionario(contr).toLowerCase().includes(valorFiltrado));
+    } else return [];
+  }
+
+  private loadDependencias(text: string) {
+    const queryOptions$ = text.length > 3 ?
+      this.oikosHelper.getDependencias(text) :
+      new Observable((obs) => { obs.next([{}]); });
+    return combineLatest([queryOptions$]).pipe(
+      map(([queryOptions_$]) => ({
+        queryOptions: queryOptions_$,
+      })),
+    );
   }
 
-  // MÉTODO PARA FILTRAR PROVEEDORES
-  changeNombreProveedor(event) {
-    this.proveedorfiltro = event.target.value;
-    this.loadProveedoresElementos();
+  private loadFuncionarios(): Promise<void> {
+    return new Promise<void>(resolve => {
+      this.tercerosHelper.getTercerosByCriterio('funcionarios').toPromise().then(res => {
+        this.Funcionarios = res;
+        resolve();
+      });
+    });
+  }
 
+  private validarTercero(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const valor = control.value;
+      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '' ? true : false;
+      const checkInvalidString = typeof (valor) === 'string' && valor !== '' ? true : false;
+      const checkInvalidTercero = typeof (valor) === 'object' && !valor.Tercero ? true : false;
+      return checkStringLength ? { errorLongitudMinima: true } :
+        checkInvalidString || checkInvalidTercero ? { terceroNoValido: true } : null;
+    };
   }
-  // MÉTODO PARA VERIFICAR QUE EL DATO EN EL INPUT CORRESPONDA CON UNO EN LA LISTA
-  verfificarProveedor() {
-    if (this.elementos.length !== 0) {
-
-      if (this.proveedorfiltro !== '') {
-        if (!this.elementos.find(element => element === this.proveedorfiltro)) {
-          this.proveedorfiltro = '';
-          this.pUpManager.showErrorAlert('El contrato seleccionado no existe!');
-        } else {
-          this.onSubmit();
-        }
-      }
-    }
+
+  private validateObjectCompleter(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const valor = control.value;
+      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '' ? true : false;
+      const checkInvalidString = typeof (valor) === 'string' && valor !== '' ? true : false;
+      const checkInvalidObject = typeof (valor) === 'object' && !valor.Id ? true : false;
```

Comment:
Uso innecesario de ternas
```suggestion
      const checkStringLength = typeof (valor) === 'string' && valor.length < 4 && valor !== '';
      const checkInvalidString = typeof (valor) === 'string' && valor !== '';
      const checkInvalidObject = typeof (valor) === 'object' && !valor.Id;
```
Nota: Cuando a futuro se requiera negar, agrupar todo entre paréntesis y colocar el operador `!` al comienzo